src/main/scripts/org/reaktivity/specification/nukleus/kafka/streams/fetch/filter.age.historical/client.rpt

```diff
@@ -0,0 +1,95 @@
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+
+property clientAddress "nukleus://streams/kafka#0"
+
+connect await ROUTED_CLIENT
+        ${clientAddress}
+    option nukleus:window 8192
+    option nukleus:transmission "half-duplex"
+
+write nukleus:begin.ext ${kafka:beginEx()
+                               .typeId(nukleus:id("kafka"))
+                               .meta()
+                                   .topic("test")
+                                   .build()
+                               .build()}
+
+connected
+
+read nukleus:begin.ext ${kafka:beginEx()
+                              .typeId(nukleus:id("kafka"))
+                              .meta()
+                                  .topic("test")
+                                  .build()
+                              .build()}
+
+read nukleus:data.ext ${kafka:dataEx()
+                             .typeId(nukleus:id("kafka"))
+                             .meta()
+                                 .partition(0, 177)
+                                 .build()
+                             .build()}
+
+read notify ROUTED_BROKER_CLIENT
+
+connect await ROUTED_BROKER_CLIENT
+        ${clientAddress}
+    option nukleus:window 8192
+    option nukleus:transmission "half-duplex"
+    option nukleus:affinity 0xb1
+
+write nukleus:begin.ext ${kafka:beginEx()
+                               .typeId(nukleus:id("kafka"))
+                               .fetch()
+                                   .topic("test")
+                                   .partition(0, 1)
+                                   .filter()
+                                       .age("HISTORICAL")
+                                       .build()
+                                   .build()
+                               .build()}
+
+connected
+
+read nukleus:begin.ext ${kafka:beginEx()
+                              .typeId(nukleus:id("kafka"))
+                              .fetch()
+                                  .topic("test")
+                                  .partition(0, 1, 2)
+                                  .build()
+                              .build()}
+
+read nukleus:data.ext ${kafka:matchDataEx()
+                             .typeId(nukleus:id("kafka"))
+                             .fetch()
+                                 .partition(0, 1, 2)
+                                 .key("key1")
+                                 .header("header2", "value2")
+                                 .build()
+                             .build()}
+read "Hello, world"
+
+read nukleus:data.ext ${kafka:matchDataEx()
+                             .typeId(nukleus:id("kafka"))
+                             .fetch()
+                                 .partition(0, 2, 2)
+                                 .header("header1", "value1")
+                                 .build()
+                             .build()}
+read "Hello, world"
+
+read notify SEND_MESSAGE_3
```

Comment:
Suggest changing this to `RECEIVED_MESSAGE_2` instead, to avoid coupling between scripts, then `awaitBarrier("RECEIVED_MESSAGE_2")` in the IT and `notifyBarrier("SEND_MESSAGE_3"), since the IT is what brings these scripts together.

Also, we need to add `read closed` here to verify that the stream is closed after all historical messages are delivered, since no new messages can become historical at that point.
```
read closed
write close
```